homeassistant/components/backup/config.py

```diff
@@ -62,16 +122,150 @@ async def load(self) -> None:
         if stored:
             self.data = BackupConfigData.from_dict(stored)
 
-    async def save(self) -> None:
+        self.data.schedule.apply(self._manager)
+
+    @callback
+    def save(self) -> None:
         """Save config."""
-        await self._store.async_save(self.data.to_dict())
+        self._store.async_delay_save(self._data_to_save, DELAY_SAVE)
+
+    @callback
+    def _data_to_save(self) -> StoredBackupConfig:
+        """Return data to save."""
+        return self.data.to_dict()
 
     async def update(
-        self, *, max_copies: int | None | UndefinedType = UNDEFINED
+        self,
+        *,
+        addons_included: list[str] | None | UndefinedType = UNDEFINED,
+        agent_ids: list[str] | UndefinedType = UNDEFINED,
+        database_included: bool | UndefinedType = UNDEFINED,
+        folders_included: list[str] | None | UndefinedType = UNDEFINED,
+        max_copies: int | None | UndefinedType = UNDEFINED,
+        name: str | None | UndefinedType = UNDEFINED,
+        password: str | None | UndefinedType = UNDEFINED,
+        schedule: ScheduleParameterDict | UndefinedType = UNDEFINED,
     ) -> None:
         """Update config."""
-        for param_name, param_value in {"max_copies": max_copies}.items():
+        if schedule is not UNDEFINED:
+            new_schedule = replace(self.data.schedule, **schedule)
+            if new_schedule != self.data.schedule:
+                self.data.schedule = new_schedule
+                self.data.schedule.apply(self._manager)
+
+        for param_name, param_value in {
+            "addons_included": addons_included,
+            "agent_ids": agent_ids,
+            "database_included": database_included,
+            "folders_included": folders_included,
+            "max_copies": max_copies,
+            "name": name,
+            "password": password,
+        }.items():
             if param_value is not UNDEFINED:
                 setattr(self.data, param_name, param_value)
 
-        await self.save()
+        self.save()
+
+
+@dataclass(kw_only=True)
+class BackupSchedule:
+    """Represent the backup schedule."""
+
+    daily: bool
+    never: bool
+    weekday: str | None
+
+    def __post_init__(self) -> None:
+        """Validate the schedule."""
+        if (
+            not self.never
+            and self.daily
+            and self.weekday
+            or not self.never
+            and not self.daily
+            and not self.weekday
+            or self.never
+            and (self.daily or self.weekday)
```

Comment:
This is very hard to read, can we group the statements with parentheses to make the three states we allow for clearer?

Reply:
The 9-member enum idea might be better